staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -1102,13 +1304,20 @@ func emitCallsToValidators(c *generator.Context, validations []validators.Functi
 				sw.Do("]", nil)
 			}
 			sw.Do("(fldPath, $.deref$obj", targs)
+			if v.Flags().IsSet(validators.UpdateOnly) {
```

Comment:
It might just a naming/docs problem.  My intent is:

- validators.UpdateOnly (flag) - Declares the a validation function *offers* update validation (it has an old value argument and is designed to validate a change to a field)
- `isUpdate` - Tracks what sort of request is being handled (create or update)
- `oldVal/oldObj` - For validator plugin function calls,  this provides the old value, which is only nil if there was no correlated old value (field was unset or there was no corresponding map entry).  For traversal calls, this can be nil either when `isUpdate=false` or when there is no correlated old value.

Reply:
Oops, what you're suggesting is much simpler... I've updated to use the above var.